crates/compiler/can/src/desugar.rs

```diff
@@ -485,10 +488,196 @@ pub fn desugar_expr<'a>(
+                        field_names.push(loc_name);
+                        field_vals.push(loc_val);
+                    }
+                    AssignedField::LabelOnly(loc_name) => {
+                        field_names.push(loc_name);
+                        field_vals.push(arena.alloc(Loc {
+                            region: loc_name.region,
+                            value: Expr::Var {
+                                module_name: "",
+                                ident: loc_name.value,
+                            },
+                        }));
+                    }
+                    AssignedField::OptionalValue(loc_name, _, loc_val) => {
+                        return arena.alloc(Loc {
+                            region: loc_expr.region,
+                            value: OptionalFieldInNewRecordBuilder(arena.alloc(loc_name), loc_val),
+                        });
+                    }
+                    AssignedField::SpaceBefore(_, _) | AssignedField::SpaceAfter(_, _) => {
+                        unreachable!("Should have been desugared in `desugar_field`")
+                    }
+                    AssignedField::Malformed(_name) => {}
+                }
+            }
+
+            let closure_arg_from_field = |field: Loc<&'a str>| Loc {
+                region: field.region,
+                value: Pattern::Identifier {
+                    ident: arena.alloc_str(&format!("#{}", field.value)),
+                },
+            };
+
+            let combiner_closure = arena.alloc(Loc::at_zero(Closure(
+                arena.alloc_slice_copy(&[
+                    Loc::at_zero(Pattern::Identifier {
+                        ident: "#record_builder_closure_arg_a",
+                    }),
+                    Loc::at_zero(Pattern::Identifier {
+                        ident: "#record_builder_closure_arg_b",
+                    }),
+                ]),
+                arena.alloc(Loc::at_zero(Tuple(Collection::with_items(
+                    Vec::from_iter_in(
+                        [
+                            &*arena.alloc(Loc::at_zero(Expr::Var {
+                                module_name: "",
+                                ident: "#record_builder_closure_arg_a",
+                            })),
+                            &*arena.alloc(Loc::at_zero(Expr::Var {
+                                module_name: "",
+                                ident: "#record_builder_closure_arg_b",
+                            })),
+                        ],
+                        arena,
+                    )
+                    .into_bump_slice(),
+                )))),
+            )));
+
+            let closure_args = {
+                if field_names.len() <= 2 {
+                    arena.alloc_slice_copy(&[
+                        closure_arg_from_field(field_names[0]),
+                        closure_arg_from_field(field_names[1]),
+                    ])
```

Comment:
We would take this branch if there were 2 fields (I use <= 2 as a defensive programming practice), and I do this because we don't have to add the tuple closure anywhere, just the record building closure. However, if this could be written to not need the `<=` guard statement here, that would be more clear. Let me try to consolidate.

Reply:
Consolidation just made it less readable, so I didn't do that. I did however change this to `if len() == 2` for clarity.